custodial-copy-backend/src/test/scala/uk/gov/nationalarchives/custodialcopy/MainTest.scala

```diff
@@ -196,14 +246,65 @@ class MainTest extends AnyFlatSpec with MockitoSugar with EitherValues {
 
   "runCustodialCopy" should "not call the process method if no messages are received" in {
     val processor = mock[Processor]
-    when(processor.process(any[MessageResponse[Option[ReceivedSnsMessage]]])).thenReturn(IO.unit)
-    val utils = new MainTestUtils(typesOfSqsMessages = Nil, objectVersion = 0)
+    when(processor.process(any[MessageResponse[Option[ReceivedSnsMessage]]], ArgumentMatchers.eq(false))).thenReturn(IO.unit)
+    val utils = new MainTestUtils(typesOfSqsMsgAndDeletionStatus = Nil, objectVersion = 0)
 
     runCustodialCopy(utils.sqsClient, utils.config, processor)
 
-    verify(processor, never()).process(any[MessageResponse[Option[ReceivedSnsMessage]]])
+    verify(processor, never()).process(any[MessageResponse[Option[ReceivedSnsMessage]]], any[Boolean])
+  }
+
+  "runCustodialCopy" should "(given a CO message with 'deleted' set to 'true') throw an Exception" in {
+    val fixity = Fixity("SHA256", "")
+
+    val utils = new MainTestUtils(
+      List((ContentObject, true)),
+      typesOfMetadataFilesInRepo = List(InformationObject, ContentObject),
+      objectVersion = 3,
+      fileContentToWriteToEachFileInRepo = List("fileContent1"),
+      entityDeleted = true
+    )
+
+    val repo = utils.repo
+
+    utils.latestObjectVersion(repo, utils.ioId) must equal(3)
+
+    val err: Throwable = getError(utils.sqsClient, utils.config, utils.processor)
+
+    err.getMessage must equal(s"Content Object '${utils.coId1}' has been deleted")
   }
 
+  "runCustodialCopy" should "(given a CO and IO message that both have 'deleted' set to 'true') parse the CO message first, " +
```

Comment:
Why does it parse the CO message first? If they're both deleted they can be processed in any order can't they?

Reply:
I think it was to verify and ensure that the CO messages get processed first, as the CO Exception being thrown for the CO is proof. I was going to do an add/update of the CO (which was the reason why we decided to process COs first) but I couldn't (at the time) think of an easy way to prove/check that the  CO was processed first in the test

Reply:
Sorry, I think I was tired when I wrote that message, I don't think it was to ensure that the CO gets run first, it's to ensure that the IO message is still processed even though the CO one threw an error. I think I was also meant to add a check to see if non-deleted messages get processed first, but then I somehow merged the test names and created a hybrid. I will change the test name and add a test to ensure non-deleted messages run first.